test/fail_compilation/diag20059.d

```diff
@@ -0,0 +1,15 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/diag20059.d(14): Error: mismatched function return type inference of `string[]` and `string` (inferred at fail_compilation/diag20059.d(12))
```

Comment:
Use `errorSupplemental`, to make the error message a bit easier to digest:

```suggestion
fail_compilation/diag20059.d(14): Error: Expected return type of `string`, but got `string[]`.
fail_compilation/diag20059.d(12):    Return type of `string` inferred here.
```

Reply:
Done.